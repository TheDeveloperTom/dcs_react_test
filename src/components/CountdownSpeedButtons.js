import React from "react";

export default ({ setActiveSpeed, speedOptions = [1, 1.5, 2] }) => (
  <>
    {speedOptions.map(speed => (
      <button key={speed} onClick={() => setActiveSpeed(speed)}>
        {speed}x
      </button>
    ))}
  </>
);
