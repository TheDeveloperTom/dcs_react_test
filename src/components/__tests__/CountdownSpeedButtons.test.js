import React from "react";
import CountdownSpeedButtons from "../CountdownSpeedButtons";

describe("CountdownSpeedButtons", () => {
  const props = {
    speedOptions: [1, 1.5, 2],
    setActiveSpeed: jest.fn()
  };
  const component = mount(<CountdownSpeedButtons {...props} />);

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("renders correct number of speed buttons", () => {
    expect(component.find("button")).toHaveLength(props.speedOptions.length);
  });

  it("renders correct text inside buttons", () => {
    expect(component.first("button").text()).toContain(
      `${props.speedOptions[0]}x`
    );
  });

  it("invokes a given callback on buttons click", () => {
    component.first("button").simulate("click");
    expect(props.setActiveSpeed).toHaveBeenCalled();
  });
});
