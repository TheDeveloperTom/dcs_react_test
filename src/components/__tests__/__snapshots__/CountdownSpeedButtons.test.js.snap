// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountdownSpeedButtons renders properly 1`] = `ReactWrapper {}`;
